curvefs/src/metaserver/copyset/copyset_node.cpp

```diff
@@ -333,15 +355,13 @@ void CopysetNode::on_snapshot_save(braft::SnapshotWriter* writer,
 
     writer->add_file(kConfEpochFilename);
 
-    std::string metaDataFile = writer->get_path() + "/" + kMetaDataFilename;
-
     // TODO(wuhanqing): MetaStore::Save will start a thread and do task
     // asynchronously, after task completed it will call
     // OnSnapshotSaveDoneImpl::Run
     // BUT, this manner is not so clear, maybe it better to make thing
     // asynchronous directly in here
-    metaStore_->Save(metaDataFile, new OnSnapshotSaveDoneClosureImpl(
-                                       this, writer, done, metricCtx));
+    metaStore_->Save(writer->get_path(), new OnSnapshotSaveDoneClosureImpl(
```

Comment:
ApplyIndex does not record after `applyQueue_->Flush();`?
Or the LogIndex does not code in kv?

Reply:
snapshot is serial with other requests, so it already has an implicit apply index in it, and in the future, I think maybe we can use apply index to avoid blocking other requests when doing snapshot, but it's not as easy as putting the apply index into storage in here.